File: oop.js
// Programacion Orientada a Objetos
/* 

- Abstraccion
- Encapsulamiento
- Herencia
- Polimorfismo

*/

person = {
    name: '',
    lastname: '',
    age: 0,
    eat: function(){}
}

class Person {
    name = ""
    lastname = ""
    age = 0

    constructor(name, lastname){
        this.name = name || "John";
        this.lastname = lastname || "Doe";
        this.age = "Unknown"
    }

    eat(){
        console.log("Metodo Eat");
    }

    fullname(){
        return `${this.name} ${this.lastname}`
    }
}


let est1 = new Person("Alex", "Cabrera");
//est1.name = "Alex"
//est1.lastname = "Cabrera"
est1.age = 30 

est1.eat()

console.log(est1.fullname());


class Student extends Person {

}

class Professor extends Person {

}


est2 = new Student("Jane", "Doe");
prof1 = new Professor("Albert", "Smith")

console.log(est2.fullname());
console.log(prof1.fullname());

File: oop.py
# Programacion Orientada a Objetos
""" 

- Abstraccion
- Encapsulamiento
- Herencia
- Polimorfismo

"""

person = {
    "name": "",
    "lastname": "",
    "age": 0,
    "eat": lambda x: print(x)
}

class Person:
    name = ""
    lastname = ""
    age = ""

    def __init__(self, name = "John", lastname = "Doe"):
        self.name = name
        self.lastname = lastname
        self.age = "Unknown"

    def eat(self):
        print("Metodo Eat")

    def saludo(self):
        return self.fullname()

    def fullname(self):
        return f"{self.name} {self.lastname}"


est1 = Person()

print(est1.fullname())

class Student(Person):
    
    def fullname(self):
        return f"Estudiante: {self.name} {self.lastname}"

    def eat(self, meal):
        print(f"Estoy comiendo: ", meal)


est2 = Student("Anna", "Smith")
print(est2.eat("Pollo"))


class A:
    itemA = "itemA"
    def uno(self):
        return "1A"

class B:
    itemB = "itemB"
    def uno(self):
        return "1B"

class C(A, B):
    pass


c1 = C()
print(c1.itemA)